src/main/java/org/datadog/jenkins/plugins/datadog/clients/DatadogHttpClient.java

```diff
@@ -87,35 +90,25 @@ of this software and associated documentation files (the "Software"), to deal
      */
     @SuppressFBWarnings(value="DC_DOUBLECHECK")
     public static DatadogClient getInstance(String url, String logIntakeUrl, Secret apiKey){
-        if(enableValidations){
-            if (url == null || url.isEmpty()) {
-                logger.severe("Datadog Target URL is not set properly");
-                return null;
-            }
-            if (apiKey == null || Secret.toString(apiKey).isEmpty()){
-                logger.severe("Datadog API Key is not set properly");
-                return null;
-            }
-            if (logIntakeUrl == null || logIntakeUrl.isEmpty()){
-                logger.warning("Datadog Log Intake URL is not set properly");
-            }
-        }
-
-        if(instance == null){
-            synchronized (DatadogHttpClient.class) {
-                if(instance == null){
-                    instance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+        // If the configuration has not changed, return the current instance without validation
+        // since we've already validated and/or errored about the data
+        if (lastInstance != null && (StringUtils.equals(lastInstance.getLogIntakeUrl(), logIntakeUrl))
+        && (StringUtils.equals(lastInstance.getUrl(), url)
+        && ((lastInstance.getApiKey() == null && apiKey == null) || lastInstance.getApiKey().equals(apiKey)))){
+            return instance;
+        } else {
+            DatadogHttpClient.lastInstance = new DatadogHttpClient(url, logIntakeUrl, apiKey);
+            if (enableValidations) {
+                try {
+                    validateCongiguration(url, logIntakeUrl, apiKey);
+                } catch(Exception e){
+                    logger.severe(e.getMessage());
+                    return null;
                 }
             }
+            instance = lastInstance;
```

Comment:
I don't see the need to store both instance and lastInstance.

We could just store `instance` of type DatadogClient, create a `newInstace` as a local variable of that method. And if newInstances is validated, replace instance by newInstance

Reply:
I agree lastInstance is not the cleanest. The issue I ran into without it is when we check for the same configuration in the beginning and instance is null (wither last time we performed validation it failed or it's the first time validating), there's no way to tell what the last parameters are because instance is null.

Reply:
Updated without lastinstance!